### ISSUE TITLE
FNX3-15521 ⁃ Make close tab button labels unique

### ISSUE THREAD

**Issue by eeejay** [mozilla-mobile/fenix#1383]
From Accessibility Scanner:

> org.mozilla.fenix.debug:id/close_tab_button
This clickable item's speakable text: "Close tab" is identical to that of 1 other item(s).

The solution requires the close button contentDescription to include the name of the tab in it. For example "close Wikipedia tab".

![image](https://user-images.githubusercontent.com/5764/55577220-ea3f2c00-56c7-11e9-9839-06a2f53b92c4.png)

**Comment by Mugurell** [comment]
I think it would still be possible to have the same name for multiple tabs if using just the page title.

**Comment by mheubusch** [comment]
not sure what advice is still needed for this issue.  my recommendation is to use "Close <title> tab" where title is the text in the site meta tag for title.  I realize, per Mugurell, that this could result in confusion if a site does not create unique titles for each page, but adding another element, like the description text, may not solve this if a site repurposes a single description too.  I'd rather err on the side of a simpler solution.

**Comment by mcarare** [comment]
@mheubusch Do you need "Close <title> tab" or "Close tab <title>" is good enough? The former will require a string change.

**Comment by mheubusch** [comment]
Actually, I think Close tab <title> is fine, @mcarare - Thank you.

**Comment by mcarare** [comment]
@mheubusch While working on this I noticed "close tab" is also used for the close button of tabs in collections. We may want to change that content description too, to include tab title. But maybe "remove tab" would be more suitable? Cause it actually removes the tab from a collection.

**Comment by mheubusch** [comment]
Yes - thank you for noticing that.  +Jeff Pfaller <jpfaller@mozilla.com>
for reference.

**Comment by jpfaller** [comment]
I agree, Remove is more appropriate. Do we need to say "tab" here?

**Comment by mheubusch** [comment]
I think tab is more specific for screen readers and might help?

**Comment by jpfaller** [comment]
Good point - let's go with Remove tab

**Comment by abodea** [comment]
I can still reproduce this issue using the latest Nightly 10/15 for `www.google.com` and the title is not working as expected for many of the websites.
I synced up with @mcarare and he agreed that he needs to investigate the title tab problem.
Based on my comment I will remove the qa:needed until the bug is fully fixed.

**Comment by AndiAJ** [comment]
Hi, I've checked this issue is still reproducible on the latest Nightly Build #12960607 using a Google Pixel 3a (Android 9) phone.

| Website  | Status|
| ------------- | ------------- |
| https://www.theverge.com/ | ✔️ |
| https://www.90min.com/ | ✔️ |
| https://h5.m.taobao.com/?sprefer=sypc00 | ✔️ |
| https://m.weibo.cn/?jumpfrom=weibocom | ✔️ |
| https://www.amazon.com/ | ❌ |
| https://m.facebook.com/ | ✔️ |
| https://www.digitaltrends.com/ | ✔️ |
| https://www.netflix.com/ro-en/ | ✔️ |
| https://www.instagram.com/ | ✔️ |
| https://www.snapchat.com/ | ✔️ |
| about:crashes | ✔️ |
| about:config | ✔️ |
| http://mediamarkt.com/ | ✔️ |
| https://mobile.willhaben.at/ | ✔️ |
| https://www.google.com/?gws_rd=ssl#spf=1571815422823 | ❌ |
| https://mobile.twitter.com/Cristiano? | ✔️ |
| https://m.soundcloud.com/ | ✔️ |
| https://www.spotify.com/ro/ | ✔️ |
| https://music.youtube.com/ | ✔️ |
| https://github.com/ | ✔️ |

► Open tabs
○ Google
![rsz_screenshot_20191023-104028](https://user-images.githubusercontent.com/51314259/67374631-21086100-f58a-11e9-94a0-24f1aff1f53a.png)


○ Amazon
![rsz_screenshot_20191023-104036](https://user-images.githubusercontent.com/51314259/67374482-e7375a80-f589-11e9-9ceb-900e650651dc.png)



► Share section
○ Google
![rsz_screenshot_20191023-104100](https://user-images.githubusercontent.com/51314259/67374149-709a5d00-f589-11e9-8191-9bd82473a972.png)

○ Amazon
![rsz_screenshot_20191023-104050](https://user-images.githubusercontent.com/51314259/67374348-b9eaac80-f589-11e9-9206-8a9e2a1c7d8e.png)

@mcarare -  It seems to be caused by the fact that the above mentioned tabs don't have a description.
P.S. Weibo and Taobao worked properly after changing the language to Simplified Chinese (China).

I'll remove the QA needed label until further clarifications.

**Comment by mcarare** [comment]
If the tab doesn't have a specific visual description it would be hard to have a content description one. I think we should open a different issue (if there isn't one already) regarding the tab title not updating.  IMO the content description is fixed, in the sense that it reflects the visual description.

**Comment by boek** [comment]
@mcarare ^^ :)

**Comment by mcarare** [comment]
@boek Concerning the content description there is no improvement possible, as long as tab title does not load properly either. Another change would be to have just the site url and another identificator (number?) but I don't think it would be a valuable infor for talkback users.
 @eeejay what do you think?

**Comment by liuche** [comment]
@MReschenberg could you advise here? What would be a reasonable thing to do in this case, when there is no title in the tab?

**Comment by MReschenberg** [comment]
@liuche When you say "no title in tab" do you mean the page has no `<title>...</title>` styling? 
On desktop, tabs are identified to screen readers by their title styling. When a title isn't specified, the page is identified by its URL, this seems appropriate in this case too (though verbose).

**Comment by liuche** [comment]
Thanks for the quick reply! Yes, for pages that don't have a `<title>...</title>` styling. (In this particular case, I suspect this is a Fenix bug, rather than the page not having a title.)

What we could try here @mcarare is reading the url (or just the domain, if that is straightforward). Is there a bug filed for missing titles?

**Comment by mcarare** [comment]
@liuche https://github.com/mozilla-mobile/fenix/issues/5351 is similar to this issue.

**Comment by mcarare** [comment]
For QA: Please retest if the issues in https://github.com/mozilla-mobile/fenix/issues/1383#issuecomment-545338897 are still reproducible.

**Comment by abodea** [comment]
Verified as fixed on the latest Nightly **12/18** with **Google Pixel 3a XL(Android 10)** for more details please check the following:

| Website  | Status|
| ------------- | ------------- |
| https://www.theverge.com/ | ✔️ |
| https://www.90min.com/ | ✔️ |
| https://h5.m.taobao.com/?sprefer=sypc00 | ✔️ |
| https://m.weibo.cn/?jumpfrom=weibocom | ✔️ |
| https://www.amazon.com/ | ✔️ |
| https://m.facebook.com/ | ✔️ |
| https://www.digitaltrends.com/ | ✔️ |
| https://www.netflix.com/ro-en/ | ✔️ |
https://www.google.com/?gws_rd=ssl#spf=1571815422823 | ✔️|

All the verified websites without a description worked as expected.
Based on my comment I will close this issue.

**Comment by Mugurell** [comment]
Closing as per https://github.com/mozilla-mobile/fenix/issues/1383#event-2893027262
